pkg/theia/commands/policy_recommendation_list.go

```diff
@@ -0,0 +1,162 @@
+			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+			Namespace(flowVisibilityNS).
+			Resource("sparkapplications").
+			Do(context.TODO()).Into(sparkApplicationList)
+		if err != nil {
+			return err
+		}
+
+		completedPolicyRecommendationList, err := getCompletedPolicyRecommendationList(clientset, kubeconfig, endpoint, useClusterIP)
+
+		if err != nil {
+			return err
+		}
+
+		sparkApplicationTable := [][]string{
+			{"CreateTime", "CompleteTime", "ID", "Status"},
+		}
+		idMap := make(map[string]bool)
+		for _, sparkApplication := range sparkApplicationList.Items {
+			id := sparkApplication.ObjectMeta.Name[3:]
+			idMap[id] = true
+			sparkApplicationTable = append(sparkApplicationTable,
+				[]string{
+					FormatTimestamp(sparkApplication.ObjectMeta.CreationTimestamp.Time),
+					FormatTimestamp(sparkApplication.Status.TerminationTime.Time),
+					id,
+					strings.TrimSpace(string(sparkApplication.Status.AppState.State)),
+				})
+		}
+
+		for _, completedPolicyRecommendation := range completedPolicyRecommendationList {
+			if _, ok := idMap[completedPolicyRecommendation.id]; !ok {
+				idMap[completedPolicyRecommendation.id] = true
+				sparkApplicationTable = append(sparkApplicationTable,
+					[]string{
+						"N/A",
+						FormatTimestamp(completedPolicyRecommendation.timeComplete),
+						completedPolicyRecommendation.id,
+						"COMPLETED",
+					})
+			}
+		}
+
+		TableOutput(sparkApplicationTable)
+		return nil
+	},
+}
+
+func getCompletedPolicyRecommendationList(clientset kubernetes.Interface, kubeconfig string, endpoint string, useClusterIP bool) (completedPolicyRecommendationList []policyRecommendationRow, err error) {
+	connect, portForward, err := setupClickHouseConnection(clientset, kubeconfig, endpoint, useClusterIP)
+	if portForward != nil {
+		defer portForward.Stop()
+	}
+	if err != nil {
+		return completedPolicyRecommendationList, err
+	}
+	query := "SELECT timeCreated, id FROM recommendations;"
+	rows, err := connect.Query(query)
+	if err != nil {
+		return completedPolicyRecommendationList, fmt.Errorf("failed to get recommendation jobs: %v", err)
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var row policyRecommendationRow
+		err := rows.Scan(&row.timeComplete, &row.id)
+		if err != nil {
+			return completedPolicyRecommendationList, fmt.Errorf("err when scaning recommendations row %v", err)
+		}
+		completedPolicyRecommendationList = append(completedPolicyRecommendationList, row)
+	}
+	return completedPolicyRecommendationList, nil
+}
+
+func init() {
+	policyRecommendationCmd.AddCommand(policyRecommendationListCmd)
+	policyRecommendationListCmd.Flags().String(
+		"clickhouse-endpoint",
+		"",
+		"The ClickHouse service endpoint.",
+	)
+	policyRecommendationListCmd.Flags().Bool(
+		"use-cluster-ip",
+		false,
+		`Enable this option will use Service ClusterIP instead of port forwarding when connecting to the ClickHouse service.
+It can only be used when running theia in cluster.`,
+	)
```

Comment:
Not sure whether it would be better to put this into global flags, as I think up to now all commands requires this flag.

Reply:
I think both `clickhouse-endpoint` and `use-cluster-ip` might be added to glabal flags?

Reply:
Hi Yanjun and Yuntang, yes there are some duplicates here, but `status` command uses `use-cluster-ip` only and is used for `Spark Monitoring Service` instead of `ClickHouse Service` in other commands.
Ultimately, I think the connection to `Spark Monitoring Service` and `ClickHouse Service` should be handled by the middle layer app theia manager. We could remove these flags then imo.

Reply:
I think for ClickHouse service, we also need these two flags in order to setup clickhouse connection? Currently, I add these two flags when using `theia clickhouse status` in PR #59

Reply:
For our `theia pr status` command, we're fetching status from k8s api server and spark monitoring service, that's why it doesn't need to connect with clickhouse for now.

But I also realize that we could improve the `status` command to let it can check completed jobs in clickhouse only. In that case, all commands needs connection with clickhouse, I plan to create a separate PR for this improvement of `status` command first.